Windows & Organization support

MayD524 opened the pull request:
I saw that there wasn't much support for windows and wanted to throw in some code to help out! The install.ps1 file could be shrunk, but I'd rather be a bit more verbose.

In terms of changes to the go code, just give it some feet so that users who have orgs that they are apart of can fully enjoy the power of this app!

Lmk if you have any questions/comments, love the project and I hope this little contribution helps!

MayD524 committed:
Windows & Organization support

ibigio commented on cli/cli.go:
````diff
@@ -308,11 +323,17 @@ func printAPIKeyNotSetMessage() {
 	3. Set your key by running:
-	` + "\n```bash\nexport OPENAI_API_KEY=[your key]\n```\n" + "    4. (Recommended) Add that ^ line to your .zshrc or .bashrc file.\n")
+	%s
+	4. (Recommended) Add that ^ line to your %s file(s).
````
nit: want to keep "file" singular instead of "file(s)" (since users will only change one file)

ibigio commented on install.ps1:
```diff
@@ -0,0 +1,84 @@
+
+
+# Replace these values with your tool's information:
+$REPO_OWNER = "ibigio"
```
I'm guessing there's a way to dynamically get the user? Would like to avoid people having to edit this file for install.

MayD524 replied:
We could do cli args

ibigio commented on cli/cli.go:
````diff
@@ -300,6 +301,20 @@ func initialModel(prompt string, client *openai.OpenAIClient) model {
 
 // === Main === //
 
+func getShellSyntax() string {
+	if runtime.GOOS == "windows" {
+		return "\n```powershell\n$env:OPENAI_API_KEY = \"[your key]\"\n```"
+	}
+	return "\n```bash\nexport OPENAI_API_KEY=[your key]\n```"
+}
+
+func getProfileScriptName() string {
+	if runtime.GOOS == "windows" {
+		return "$profile"
+	}
+	return ".zshrc or .bashrc"
+}
+
````
Maybe let's make these variables inside `printAPIKeyNotSetMessage()` instead of functions. Kind of like:

```go
set_api_key_syntax := [default to linux]
profile_file_syntax := [default to zsh and bash]
if runtime.GOOS == "windows" {
  [set them to windows values]
}
```

ibigio commented on cli/cli.go:
```diff
@@ -330,12 +351,14 @@ var RootCmd = &cobra.Command{
 		// join args into a single string separated by spaces
 		prompt := strings.Join((args), " ")
 		apiKey := os.Getenv("OPENAI_API_KEY")
+		// The organization key is optional ( just for us enterprise users :) )
```
Hahaha love it! If you git rid of "us" I'll keep the smiley ;D

MayD524 replied:
Okok just joking around in comments 😂

ibigio commented on cli/cli.go:
````diff
@@ -308,11 +323,17 @@ func printAPIKeyNotSetMessage() {
-	` + "\n```bash\nexport OPENAI_API_KEY=[your key]\n```\n" + "    4. (Recommended) Add that ^ line to your .zshrc or .bashrc file.\n")
+	%s
+	4. (Recommended) Add that ^ line to your %s file(s).
+	5. If you are a member of an organization do the same thing but with OPENAI_ORGANIZATION_KEY`, getShellSyntax(), getProfileScriptName())
````
I expect most users to be individuals, so to keep things clean I'd rather not add 5 here, but instead add it potentially to an error message a user would get if they don't set the org key. If you can figure out what http error that is then you could also add that yourself! Or lmk and I can as well.

MayD524 replied:
We could easily check for it. I'll do some digging.

ibigio reviewed (changes requested):
Hey May, thank you so much for the contributions! I knew I should eventually acknowledge that windows exists, but frankly didn't expect to get this much traction :) 

Actually wasn't aware OpenAI API could have org keys, this is important add to support for too so thank you.

I left a few comments, mostly just nits. Once you resolve them I can take another look and do my best to find a windows machine and test it!

(You're also the first real open source contribution any of my projects receive, so just know you made my day.)

MayD524 commented:
Hey no problem! I found your project and as a primarily windows user had to toss some stuff in as well. Also I use OpenAi's API through my org so my personal account doesn't have a card attached but my Org does so just had to a little to get it working!

No problem though! I love what you're doing and saw something that could be helped!

Will get onto the nits soon!

MayD524 committed:
Fixed nits, added CLI args to windows install, added q.exe to .gitignore

MayD524 commented:
Sorry for taking forever currently in a hackaton! But I hope like the changes!

Created new commit gh-2743c5e

Just to wrap it up changes include:
 - Added CLI args to install.ps1 (users can change targeted repo and output name
 - Added Help to install.ps1 (does install.sh have one? I haven't checked)
 - Changed phrasing "file(s)" back to "file"
 - Removed "us" from cli :,) 😂 
 - Moved all code for switching between windows/unix syntax from function to 4 lines (wow you can shorten go code?)
 - Trying to figure out where to prompt user for org key (probably if we fail due to lack of tokens?)
    - Side note on the above we  should probably have a cleaner error for that one
